packages/cosmic-swingset/src/anylogger-agd.js

```diff
@@ -0,0 +1,165 @@
+/* eslint-env node */
+import {
+  getEnvironmentOptionsList,
+  getEnvironmentOption,
+} from '@endo/env-options';
+import anylogger from 'anylogger';
+import { defineName } from '@agoric/internal/src/js-utils.js';
+import util from 'util';
+
+/** @import {BaseLevels} from 'anylogger'; */
+/** @typedef {keyof BaseLevels} LogLevel; */
+
+const VAT_LOGGER_PREFIXES = Object.freeze([
+  'SwingSet:vat',
+  'SwingSet:ls', // "ls" for "liveslots"
+]);
+
+//#region Golang zerolog-like output affordances
+
+let shortDateTimeFormatter;
+const plainTime = d => {
+  if (!shortDateTimeFormatter) {
+    shortDateTimeFormatter = new Intl.DateTimeFormat(undefined, {
+      timeStyle: 'short',
+    });
+  }
+  const shortTime = shortDateTimeFormatter.format(d).replace(/(\d) /, '$1');
+  return shortTime;
+};
+
+const levelToPlainLevel = {
+  error: 'ERR',
+  warn: 'WRN',
+  info: 'INF',
+  log: 'LOG',
+  debug: 'DBG',
+};
+
+/**
+ * @param {Date} d
+ * @returns {string}
+ */
+const dateToNumericOffsetString = d => {
+  // TODO: Use Temporal.ZonedDateTime
+  // `toString({ fractionalSecondDigits: 0 })`.`
+  const pad2 = n => (n < 10 ? `0${n}` : `${n}`);
+  const utcOffsetMinutes = -d.getTimezoneOffset();
+  const utcOffsetAbsMinutes = Math.abs(utcOffsetMinutes);
+  const utcOffsetString = `${utcOffsetMinutes < 0 ? '-' : '+'}${pad2(Math.floor(utcOffsetAbsMinutes / 60))}:${pad2(utcOffsetAbsMinutes % 60)}`;
+  const utcDateWithSameLocalTime = new Date(
+    d.getTime() + utcOffsetMinutes * 60e3,
+  );
+  return utcDateWithSameLocalTime
+    .toISOString()
+    .replace(/[.][0-9]+Z$/, utcOffsetString);
+};
+
+/**
+ * @type {Record<string,
+ *   (opts: { now: Date, label: string, level: LogLevel, args: any[] }) => any[]
+ * >}
+ */
+const logFormatterForLogFormat = {
+  legacy: ({ now, label, args }) => {
+    const timestamp = now.toISOString();
+    return [`${timestamp} ${label}:`, ...args];
+  },
+  plain: ({ now, label, level, args }) => {
+    const timestamp = plainTime(now);
+    const plainLevel = levelToPlainLevel[level];
+    return [`${timestamp} ${plainLevel} ${label}:`, ...args];
+  },
+  json: ({ now, label, level, args }) => {
+    const message = [`${label}:`, ...args]
+      .map(a => (typeof a === 'string' ? a : util.inspect(a)))
+      .join(' ');
+    return [
+      JSON.stringify({
+        level,
+        time: dateToNumericOffsetString(now),
+        message,
+      }),
+    ];
+  },
+};
+
+/** @type {string} */
+let logFormat;
+/** @type {typeof logFormatterForLogFormat[string]} */
+let logFormatter;
+
+/** @param {string} format */
+export const setLogFormat = format => {
+  logFormat = format;
```

Comment:
I'd like to see explicit acknowledgement/justification for this exceptional static mutable state.

Logging is often an expedient exception to OCap Discipline. But I'd rather that the reader didn't think we take this lightly.